pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -230,6 +270,11 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundVirtualHosts(node *mod
 		}
 	}
 
+	// This is hack to keep consistent with previous behavior.
+	if listenerPort != 80 {
+		// only select virtualServices that matches a service
+		virtualServices = selectVirtualServices(virtualServices, servicesByName)
```

Comment:
I do not think we have any test coverage of this change. If you comment it out all tests still pass. 

This also makes it hard to review the rest as I don't understand the full scope of the behavior changed by the PR, so I am not sure if this is safe or not. We would want to make sure these routes we are adding are truly never used before removing them, as we have had a lot of regressions in this config in the past

Reply:
Yup, it passes without this selection. I explained above https://github.com/istio/istio/pull/33764#issuecomment-873505981
>there are also VirtualHost built for each route as an intermediate procedure, but not added at all.

I am thinking about adding a test for this, but does not come up with a good way

Reply:
addes test

Reply:
Removing this still passes all tests I think

Reply:
Oh but we have tests for the function itself, i see